src/table.js

```diff
@@ -659,6 +663,27 @@ export function __table(source, operations) {
       Object.fromEntries(operations.select.columns.map((c) => [c, d[c]]))
     );
   }
+  if (!primitive && operations.names) {
```

Comment:
Ah, nice!

Reply:
the real fix for the original issue you found is actually in [this change in Next](https://github.com/observablehq/observablehq/pull/10009/commits/bb017ddd0d86a6e777a420281bd41a4f08feef7e)! but while i was investigating i realized we should also add a check here.

Reply:
aaah! thanks for explaining that.